core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -149,10 +149,33 @@ public TenantObject addTrustAnchor(final PublicKey publicKey, final X500Principa
         Objects.requireNonNull(publicKey);
         Objects.requireNonNull(subjectDn);
 
+        return addTrustAnchor(publicKey.getEncoded(), publicKey.getAlgorithm(), subjectDn, autoProvisioningEnabled);
+
+    }
+
+    /**
+     * Adds a trusted certificate authority to use for authenticating devices of this tenant.
+     *
+     * @param publicKey The CA's public key in encoded form.
+     * @param publicKeyAlgorithm The algorithm of the public key.
+     * @param subjectDn The CA's subject DN.
+     * @param autoProvisioningEnabled A flag indicating whether this CA may be used for automatic provisioning.
+     * @return This tenant for command chaining.
+     * @throws NullPointerException if the public key or subjectDN parameters is {@code null}.
```

Comment:
what about the algorithm? seems that we need that too, don't we? At least there seems to be no default value defined ...